### ISSUE TITLE
Fastify vs Express

### ISSUE THREAD

**Issue by mwcz** [spaship/spaship#80]
It might be worth investigating using fastify instead of express.  It has a lot of advantages and also a Red Hatter is one of the maintainers.

**Comment by sayak-sarkar** [comment]
I have been playing around with fastify for a couple of months now and from what I have seen so far, I'd love to use fastify, as the performance benefits are quite significant, especially when it comes to response times for requests. 

Apart from this I also like the fact that it supports plugin encapsulation which is a big plus from the long-term scaling up perspective. Its package size is also quite small. (83.6kb vs. express's 218.5kb minified)

**Comment by mwcz** [comment]
Once #54 is done, we'll have three REST services in SPAship.  autosync, api (formerly sync-service), and path-proxy.

The way I see it, path-proxy is the only one where performance is absolutely the top priority.  api and autosync's performance requirements are totally satisfied by express.  express may also be more than fast enough for path-proxy, but that's a big unknown for me.  path-proxy is going to get hammered, so going to something known to be very fast would be great.

Anyway, I'm on-board with fastify.  I really like how specifying a schema for JSON messages is a built-in feature.

@sayak-sarkar, why is package size important?  None of this is going to run in browser.

**Comment by mwcz** [comment]
Closing due to lack of activity.  We can still switch to fastify, but it's a low priority at the moment.
